vowpalwabbit/cb_explore_adf_common.h

```diff
@@ -113,12 +110,16 @@ inline void cb_explore_adf_base<ExploreType>::learn(
   example* label_example = CB_ADF::test_adf_sequence(examples);
   if (label_example != nullptr)
   {
+    // Notes:  Label exists so call learn()
     data._known_cost = CB_ADF::get_observed_cost(examples);
     // learn iff label_example != nullptr
     data.explore.learn(base, examples);
   }
-  else
+  else if (!examples[0]->predict_called_before_learn)
```

Comment:
Why does the reduction not know whether or not it needs to call predict first?

Reply:
I am not very happy with having to expose this.  This was part of the changes made to improve efficiency.  I need to refresh my memory about the issue.  Will provide details once I review my commits.